include/matx/transforms/solver.h

```diff
@@ -1464,4 +1464,300 @@ void eig([[maybe_unused]] OutputTensor &out, WTensor &w,
   matx::copy(out, tv.PermuteMatrix(), stream);  
 }
 
+
+
+/**
+ * Perform a SVD decomposition using the power iteration.  This version of
+ * SVD works well on small n/m with large batch.
+ *
+ *
+ * @tparam UType
+ *   Tensor or operator type for output of U eigen vectors.
+ * @tparam SType
+ *   Tensor or operator type for output of S eigen values. SType must have Rank one less than AType.
+ * @tparam VType
+ *   Tensor or operator type for output of VT eigen vectors.
+ * @tparam AType
+ *   Tensor or operator type for output of A input tensors.
+ * @tparam X0Type
+ *   Tensor or operator type for X0 initial guess in power iteration.
+ *
+ * @param U
+ *   U tensor or operator for left eigen vectors output with size "batches by m by k"
+ * @param S
+ *   S tensor or operator for eigen values output with size "batches by k"
+ * @param VT
+ *   VT tensor or operator for right eigen vectors output as VH with size "batches by k by n"
+ * @param A
+ *   Input tensor or operator for tensor A input with size "batches by m by n"
+ * @param x0
+ *   Input tensor or operator signaling the initial guess for x0 at each power iteration.  A
+ *   Random tensor of size min(n,m) is suggested.
+ * @param iterations
+ *   The number of power iterations to perform for each eigen value.  
+ * @param stream
+ *   CUDA stream
+ * @param k
+ *    The number of eigen values to find.  Default is all eigen values: min(m,n).
+ */
+template<typename UType, typename SType, typename VTType, typename AType, typename X0Type>
+void svdpi(UType &U, SType &S, VTType &VT, AType &A, X0Type &x0, int iterations,  cudaStream_t stream, index_t k=-1) {
+  MATX_NVTX_START("", matx::MATX_NVTX_LOG_INTERNAL)
```

Comment:
Does it need a name here or will it somehow just work?

Reply:
Tyler's magic take's the parent function name.  so "svdpi" in this case.